common/src/com/thoughtworks/go/domain/materials/git/GitCommand.java

```diff
@@ -231,8 +231,22 @@ public UrlArgument workingRepositoryUrl() {
         return new UrlArgument(runOrBomb(gitConfig).outputForDisplay().get(0));
     }
 
-    public static void checkConnection(UrlArgument repoUrl) {
-        commandToCheckConnection(repoUrl).runOrBomb(repoUrl.forDisplay());
+    public static void checkConnection(UrlArgument repoUrl, String branch) {
+        CommandLine commandLine = git().withArgs("ls-remote").withArg(repoUrl);
+        ConsoleResult result = commandLine.runOrBomb(repoUrl.forDisplay());
+        if(!isValidBranch(result.outputAsString(), branch)){
+            throw invalidBranchException(repoUrl.forDisplay(), branch);
```

Comment:
That flag is not supported by the version of git on centos 1.7.1.

On Mon, Apr 18, 2016, 4:07 PM Zabil C.M <notifications@github.com> wrote:

> In common/src/com/thoughtworks/go/domain/materials/git/GitCommand.java
> <https://github.com/gocd/gocd/pull/2114#discussion_r60037148>:
>
> > @@ -231,8 +231,22 @@ public UrlArgument workingRepositoryUrl() {
> >          return new UrlArgument(runOrBomb(gitConfig).outputForDisplay().get(0));
> >      }
> >
> > -    public static void checkConnection(UrlArgument repoUrl) {
> > -        commandToCheckConnection(repoUrl).runOrBomb(repoUrl.forDisplay());
> > +    public static void checkConnection(UrlArgument repoUrl, String branch) {
> > +        CommandLine commandLine = git().withArgs("ls-remote").withArg(repoUrl);
> > +        ConsoleResult result = commandLine.runOrBomb(repoUrl.forDisplay());
> > +        if(!isValidBranch(result.outputAsString(), branch)){
> > +            throw invalidBranchException(repoUrl.forDisplay(), branch);
>
> Why can't you pass the branch name while executing the command?
>
> sh-3.2$ git ls-remote https://github.com/zabil/gocd uichanges
> 241552c3dba5aa5b4e9c3b426b5aa6a44eaa383c        refs/heads/uichanges
>
> sh-3.2$ git ls-remote https://github.com/zabil/gocd --exit-code notthere
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/gocd/gocd/pull/2114/files/37b7293affb0c15c25307280f6ad176f226ee195#r60037148>
>